components/sup/src/manager/service/health.rs

```diff
@@ -196,26 +195,28 @@ impl State {
 
     /// Repeatedly runs a health check, followed by an appropriate
     /// delay, forever.
-    pub fn check_repeatedly(self) -> impl Future<Item = (), Error = ()> {
+    /// # Locking for the returned Future (see locking.md)
+    /// * `GatewayState::inner` (write)
+    pub fn check_repeatedly_gsw(self) -> impl Future<Item = (), Error = ()> {
         future::loop_fn(self, move |state| {
             // TODO (CM): If we wanted to keep track of how many times
             // a health check has failed in the past X executions, or
             // do similar historical tracking, here's where we'd do
             // it.
             let service_group = state.service_group.clone();
-            state.clone().single_iteration().then(move |res| {
-                                                if res.is_ok() {
-                                                    trace!("Health check future for {} \
-                                                            succeeded; continuing loop",
-                                                           service_group);
-                                                    Ok(Loop::Continue(state))
-                                                } else {
-                                                    trace!("Health check future for {} failed \
-                                                            failed; continuing loop.",
-                                                           service_group);
-                                                    Ok(Loop::Continue(state))
-                                                }
-                                            })
+            state.clone().single_iteration_gsw().then(move |res| {
+                                                    if res.is_ok() {
+                                                        trace!("Health check future for {} \
+                                                                succeeded; continuing loop",
+                                                               service_group);
+                                                        Ok(Loop::Continue(state))
+                                                    } else {
+                                                        trace!("Health check future for {} \
+                                                                failed failed; continuing loop.",
```

Comment:
Oops, looks like we had "failed" twice!

Now would be a good time to fix that typo :man_facepalming:

Reply:
Done in https://github.com/habitat-sh/habitat/pull/6937/commits/8b6d4a076605fb66142212838ebcccf9bc83ee18